esmvaltool/diag_scripts/weighting/weighted_temperature_graph.py

```diff
@@ -105,23 +130,43 @@ def main(cfg):
     models = read_metadata(cfg, 'short_name')['tas']
     model_data, model_data_files = read_model_data(models)
 
-    percentiles = np.array([25, 75])
+    settings = cfg['settings']
+    if isinstance(settings.get('central_estimate', 50), (int, float)):
+        central_estimate = calculate_percentiles(
+            model_data,
+            np.array([settings.get('central_estimate', 50)]),
+        )
+        central_estimate_weighted = calculate_percentiles(
+            model_data,
+            np.array([settings.get('central_estimate', 50)]),
+            weights=weights,
+        )
+    elif settings['central_estimate'] == 'mean':
+        central_estimate = model_data.mean('model_ensemble')
+        central_estimate_weighted = model_data.weighted(weights).mean(
+            'model_ensemble')
```

Comment:
I think defining the central estimate variable at the top of the block makes it a bit easier to read.

```suggestion
    settings = cfg['settings']
    central_estimate_var = settings.get('central_estimate', 50)
    if isinstance(central_estimate_var, (int, float)):
        central_estimate = calculate_percentiles(
            model_data,
            np.array([central_estimate_var]),
        )
        central_estimate_weighted = calculate_percentiles(
            model_data,
            np.array([central_estimate_var]),
            weights=weights,
        )
    elif central_estimate_var == 'mean':
        central_estimate = model_data.mean('model_ensemble')
        central_estimate_weighted = model_data.weighted(weights).mean(
            'model_ensemble')
```